client/network/src/service.rs

```diff
@@ -667,6 +670,11 @@ impl<B: BlockT + 'static, H: ExHashT> NetworkService<B, H> {
 		&self.local_peer_id
 	}
 
+	/// Returns the `KeyPair` that defined the local `PeerId`.
+	pub fn local_identity(&self) -> &libp2p::identity::Keypair {
+		&self.local_identity
+	}
```

Comment:
This probably doesn't matter too much, but I think that exposing a function to sign something with the libp2p key is better than simply exposing the private key?

Reply:
Yeah, it is somewhat a matter of taste unless the message to be signed is not typed. There is a future PR in planning where the `sc-network` crate could use a `SyncCryptoStore` for its `node_key`, that is why I was sloppy here.

Reply:
> There is a future PR in planning where the sc-network crate could use a SyncCryptoStore 

More distant, distant future and not really planned :P

Reply:
> Yeah, it is somewhat a matter of taste unless the message to be signed is not typed. There is a future PR in planning where the sc-network crate could use a SyncCryptoStore for its node_key, that is why I was sloppy here.

I know this was discussed on a channel, but I don't really see the advantage. The keys in the keystore can change over time and at any moment, but libp2p isn't capable of changing the key it uses after initialization (for good reasons, your identity is stored in the DHT and all).
I understand the idea of loading the network key at initialization from the keystore, but I don't see the advantage in passing the keystore itself to the networking.

Reply:
I agree. For now I just added a test that checks if an `app_crypto!()` we create would be compatible with `libp2p::identity` so we notice any regressions. Keeping "the" network identity in a separate folder unencrypted like now, or keeping it together with the other keys optionally encrypted with a password provided at startup is mostly a secops question.